src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -1166,7 +1170,8 @@ def build_udtf(
     comment: Optional[str] = None,
     statement_params: Optional[Dict[str, str]] = None,
     is_permanent: bool = False,
-    session=None,
+    session: "snowflake.snowpark.session.Session" = None,
+    udtf_name: str = None,
```

Comment:
type hint? Optional[str]? Prob. with spTableName change the type name for that should be propagated here.